src/System.Diagnostics.TraceSource/src/System/Diagnostics/CorrelationManager.cs

```diff
@@ -33,7 +34,13 @@ public Stack LogicalOperationStack
         {
             get
             {
-                return GetLogicalOperationStack();
+                ImmutableArray<object> elements = _idStack.Value.ToImmutableArray();
+                Stack newStack = new Stack();
+                for (int i = elements.Length-1; i >= 0; i--)
+                {
+                    newStack.Push(elements[i]);
+                }
+                return newStack;
```

Comment:
Yes -- in that case it would be dropped on the floor.

In the search I did, I noticed that the use case for this is to display it and not do push/pop from it.